libs/utils/android/benchmark.py

```diff
@@ -0,0 +1,262 @@
+        """
+        pass
+
+################################################################################
+# Private Interface
+
+    def _parseCommandLine(self):
+
+        parser = argparse.ArgumentParser(
+                description='LISA Benchmark Configuration')
+
+        # Bootup settings
+        parser.add_argument('--boot_image', type=str,
+                default=None,
+                help='Path of the Android boot.img to be used')
+
+        # Android settings
+        parser.add_argument('--android_device', type=str,
+                default=None,
+                help='Identifier of the Android target to use')
+        parser.add_argument('--android_home', type=str,
+                default=None,
+                help='Path used to configure ANDROID_HOME')
+
+        # Test customization
+        parser.add_argument('--results_dir', type=str,
+                default=None,
+                help='Results folder, '
+                     'if specified override test defaults')
+        parser.add_argument('--collect', type=str,
+                default=None,
+                help='Set of metrics to collect, '
+                     'e.g. "energy systrace_30" to sample energy and collect a 30s systrace, '
+                     'if specified overrides test defaults')
+
+        # Measurements settings
+        parser.add_argument('--iio_channel_map', type=str,
+                default=None,
+                help='List of IIO channels to sample, '
+                     'e.g. "ch0:0,ch3:1" to sample CHs 0 and 3, '
+                     'if specified overrides test defaults')
+
+        # Parse command line arguments
+        self.args = parser.parse_args()
+
+    def _getBmConf(self):
+        if self.bm_conf is None:
+            raise NotImplementedError("Override `bm_conf` attribute")
+        # Override default configuration with command line parameters
+        if self.args.android_device:
+            self.bm_conf['device'] = self.args.android_device
+        if self.args.android_home:
+            self.bm_conf['ANDROID_HOME'] = self.args.android_home
+        if self.args.results_dir:
+            self.bm_conf['results_dir'] = self.args.results_dir
+        if self.args.collect:
+            self.bm_collect = self.args.collect
+
+        if self.args.iio_channel_map:
+            try:
+                em = self.bm_conf['emeter']
+                if em['instrument'] != 'acme':
+                    raise ValueError
+                em['channel_map'] = {}
+                for ch in self.args.iio_channel_map.split(','):
+                    ch_name, ch_id = ch.split(':')
+                    em['channel_map'][ch_name] = ch_id
+                self.bm_conf['emeter'] = em
+                self.log.info('Using iio-capture channels: %s', em)
+            except:
+                # No ACME configuration required by the test
+                pass
+
+        return self.bm_conf
+
+    def _getBmKind(self):
```

Comment:
Well... to avoid the overloaded concept of Workload (we have now wlgen and android Workloads)... I'm going to update this to be "bm_name", which stands for "benchmark name". This should be generic enough to accommodate other workloads/benchmarks in a future refactoring.